scripts/cnn_variant_wdl/vcf_analysis.R

```diff
@@ -0,0 +1,218 @@
+  geom_jitter(height = 0, width = 0.1, alpha=0.6) + 
+  ggtitle(plot_title) +
+  ylab(score_label) + 
+  xlab("Event Length: - Deletions, 0 SNPs, + Insertions")
+
+p3 <- ggplot(d, aes(x=EVENTLENGTH, y=get(score_key), group=EVENTLENGTH, color=Truth_Status)) + xlim(-20, 20) + 
+  scale_color_manual(values=statusColor) +
+  geom_jitter(height = 0, width = 0.1, alpha=0.4) + 
+  geom_violin(color="grey", alpha=0) + 
+  geom_text(aes(x=EVENTLENGTH, y=14, label=EVENTLENGTH_SUM), color="grey30", size=2, angle=60) +
+  ggtitle(plot_title) +
+  ylab(score_label) + 
+  xlab("Event Length: - Deletions, 0 SNPs, + Insertions")
+
+p4 <- ggplot(d, aes(x=EVENTLENGTH, y=get(score_key), group=EVENTLENGTH, color=Truth_Status)) + xlim(-10, 10) + 
+  scale_color_manual(values=statusColor) +
+  geom_jitter(height = 0, width = 0.2, alpha=0.4) + 
+  geom_violin(color="grey", alpha=0) + 
+  geom_text(aes(x=EVENTLENGTH, y=14, label=EVENTLENGTH_SUM), color="grey30", size=3, angle=30) +
+  ylab(score_label) + 
+  xlab("Event Length: - Deletions, 0 SNPs, + Insertions")
+
+p5 <- ggplot(d, aes(x=EVENTLENGTH, y=get(score_key), group=EVENTLENGTH, color=Truth_Status)) + 
+  scale_color_manual(values=statusColor) +
+  xlim(-5, 5) + 
+  geom_jitter(height = 0, width = 0.35, alpha=0.4) + 
+  geom_violin(color="grey", alpha=0.0) + 
+  geom_text(aes(x=EVENTLENGTH, y=14, label=EVENTLENGTH_SUM), color="grey30", size=4, angle=30) +
+  ggtitle(plot_title) +
+  ylab(score_label) + 
+  xlab("Event Length: - Deletions, 0 SNPs, + Insertions")
+
+# SNP specific plots
+print("Make SNP plots.")
+snps <- subset(d, EVENTLENGTH == 0)
+my_breaks <- ifelse(bin_by_quantile, quantile(snps[[score_key]], probs = seq(0, 1, 1.0/num_bins), na.rm=T), num_bins)
+snps$QUALITY_BIN <- cut(snps[[score_key]], breaks=my_breaks, include.lowest=T, labels=F)
+snps$QUALITY_BIN_RANGE <- cut(snps[[score_key]], breaks=my_breaks, include.lowest=T)
+mine <- lapply(strsplit(sapply(levels(snps$QUALITY_BIN_RANGE), function(x) substr(x, 2, nchar(x)-1)), ","), as.numeric)
+df <- data.frame(matrix(unlist(mine), nrow=num_bins, byrow=T))
+q_means <- rowMeans(df)
+snps$QUALITY_LOD <- q_means[snps$QUALITY_BIN] 
+snps$TPR_PREDICTION <- exp(snps$QUALITY_LOD) / (1 + exp(snps$QUALITY_LOD) )
+
+x <- rowsum(snps$ONE, snps$QUALITY_BIN)
+snps$BIN_SUM <- x[snps$QUALITY_BIN]
+snps$TRANSVERSION <- as.numeric( abs(snps$TRANSITION)==0 )
+snps$TPR <- get_proportion(snps, num_bins, snps$TP, snps$QUALITY_BIN)
+ti <- get_proportion(snps, num_bins, snps$TRANSITION, snps$QUALITY_BIN)
+tv <- get_proportion(snps, num_bins, snps$TRANSVERSION, snps$QUALITY_BIN)
+snps$TI_TV <- ti/tv
+
+p6 <- ggplot(snps, aes(x=get(score_key), y=TI_TV, group=QUALITY_BIN, color=Truth_Status, shape=TRANSITION==1)) + 
+  scale_color_manual(values=statusColor) +
+  scale_shape_discrete(name='', breaks=c("TRUE", "FALSE"), labels=c("Transition", "Transversion")) +
+  geom_point() + 
+  geom_line(color="grey") + 
+  ggtitle("Transition Transversion Ratio per score bin") +
```

Comment:
Could you do a cumulative one as well?  That's often how users set a filtering threshold.

Reply:
You can skip this if it's too arduous.

Reply:
First going to try to respond to easier comments and see if tests still pass, then will take a crack at R code cleanup and cumulative Ti/Tv.